docs/payment-methods/TWINT.md

```diff
@@ -0,0 +1,91 @@
+
+## Stored Twint payments
+### Sessions
+- Include the [`storedPaymentMethodMode`](https://docs.adyen.com/api-explorer/Checkout/latest/post/sessions#request-storePaymentMethodMode) and [`shopperReference`](https://docs.adyen.com/api-explorer/Checkout/latest/post/sessions#request-storePaymentMethodMode) parameter in your [`/sessions`](https://docs.adyen.com/api-explorer/Checkout/latest/post/sessions) request.
+- The API response will contain a list of the shopper's stored payment methods.
+- For drop-in, pass the API response as explained in [the integration guide](https://docs.adyen.com/online-payments/build-your-integration/sessions-flow/?platform=Android&integration=Drop-in) and it will show the stored payment methods. 
+- For components, get the Twint `StoredPaymentMethod` from the list and pass it when creating the `TwintComponent`:
+```Kotlin
+val storedPaymentMethods = checkoutSession.sessionSetupResponse.paymentMethodsApiResponse?.storedPaymentMethods
+val storedPaymentMethod = storedPaymentMethods?.filter { it.type == PaymentMethodTypes.TWINT }
+
+val component = TwintComponent.PROVIDER.get(
+    activity = activity, // or fragment = fragment
+    checkoutSession = checkoutSession, // Should be passed only for sessions
+    storedPaymentMethod = storedPaymentMethod,
+    configuration = checkoutConfiguration,
+    componentCallback = callback,
+)
+```
+
+### Advanced
+- When a shopper chooses to pay, they will be provided with the option to store their payment details. To remove the option to store payment details, add `setShowStorePaymentField(false)` when configuring Twint.
+- The `PaymentComponentState` will contain the shopper's choice in `data.storePaymentMethod`. 
+- Include [`storePaymentMethod`](https://docs.adyen.com/api-explorer/Checkout/latest/post/payments#request-storePaymentMethod) and [`shopperReference`](https://docs.adyen.com/api-explorer/Checkout/latest/post/payments#request-shopperReference) in the [`/payments`](https://docs.adyen.com/api-explorer/Checkout/latest/post/payments) request.
+- Include the `shopperReference` in the [`/paymentMethods`](https://docs.adyen.com/api-explorer/Checkout/latest/post/paymentMethods) request and the response will contain a list of the shopper's stored payment methods.
+- For drop-in, pass the API response as explained in [the integration guide](https://docs.adyen.com/online-payments/build-your-integration/advanced-flow/?platform=Android&integration=Drop-in) and it will show the stored payment methods.
+- For components, get the Twint `StoredPaymentMethod` from the list and pass it when creating the `TwintComponent`:
+```Kotlin
+val storedPaymentMethods = paymentMethodsApiResponse?.storedPaymentMethods
+val storedPaymentMethod = storedPaymentMethods?.filter { it.type == PaymentMethodTypes.TWINT }
```

Comment:
I think we should call `firstOrNull()` instead of `filter` here since `filter` returns a list and not an object.

Reply:
Sharp 🦅